tool/embedded-binaries.ts

```diff
@@ -60,9 +61,74 @@ export async function getEmbeddedProtocol(outPath: string): Promise<void> {
 /**
  * Gets the latest version of the Dart Sass wrapper for the Embedded Compiler.
  * Throws if an error occurs.
+ *
+ * @param release - Whether to download a release version of the Embedded
+ *   Compiler or build one from source.
+ * @param version - If `release` is `true`, the version of the Embedded Compiler
+ *   to download. If it's `false`, the Git ref to check out. Defaults to
+ *   the latest available version or `master`, respectively.
  */
-export async function getDartSassEmbedded(outPath: string): Promise<void> {
-  await fs.mkdir(outPath, {recursive: true});
+export async function getDartSassEmbedded(
+  outPath: string,
+  {
+    release = false,
+    version,
+  }: {release: boolean | undefined; version: string | undefined}
+): Promise<void> {
+  if (release) {
+    await downloadDartSassEmbedded(outPath, version);
+  } else {
+    await buildDartSassEmbedded(outPath, version);
+  }
+}
+
+// Clones the Embedded Dart Sass repo and build its executable from source.
+async function buildDartSassEmbedded(
```

Comment:
Will this ever be called when a user downloads the npm package? My only concern is requiring Node users to depend on a Dart build step instead of just downloading the release binary.

Reply:
I was only thinking about this from a development perspective. Are you expecting that this will also be run when a user installs the package? Looking at the npm docs, it looks like that's typically done in a separate postinstall script.

Reply:
I was thinking the postinstall script would also call this helper function. But, it also could make sense to just write separate logic for the postinstall

Reply:
If it calls it as a helper function, it can just pass in `release: true` to ensure that the release versions are downloaded, right?

Reply:
ah, that's right